minidump-processor/src/symbols.rs

```diff
@@ -327,4 +444,31 @@ mod symbols_shim {
         /// If we tried to parse the symbols, but failed.
         pub corrupt_symbols: bool,
     }
+
+    // These suppliers are entriely private to the implementation, so do whatever you
+    // want with them.
+
+    struct HttpSymbolSupplier {}
```

Comment:
I think the lib part of `minidump-processor`—the `process_minidump` API—should stay generic over the traits, but I don't think it should necessarily be a goal to make `minidump-stackwalk` itself able to plug in alternate implementations of those traits, do you? Its commandline arguments are pretty tightly bound to the current implementation strategy, and anyone (such as Sentry) who's going to use this crate with alternate implementations is assuredly going to be using the crate as a library and not invoking `minidump-stackwalk` directly.

We can just document that the `minidump-stackwalk` binary here is tailored to Mozilla's needs, and so it intentionally only supports fetching Breakpad-format symbol files from Tecken and caching them on local disk. (Specific implementation details subject to change in the future if Mozilla's needs change, naturally.)

Reply:
Yeah this is just like this to allow someone to prototype a symbolic-based implementation with minimal refactoring. Changing the interface is totally fine. This just lets them not have to worry about it if they don't want to.

Reply:
So to be clear, in the long-term I would just remove the feature and make symbolic-syms the only impl.

Reply:
100% on board with that plan.